index.d.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * A ComponentStartData contains the information necessary
```

Comment:
I noticed that there are a few "A {object name} contains" instances in this file. Would it make more sense to write "A {object name} instance contains"?

Reply:
Yeah, I'll rework the naming for all of these and fix up the wording.